src/App/App.js

```diff
@@ -13,16 +15,27 @@ class App extends Component {
     this.state = {
       movies: [],
       error: '',
-      currentUser: null
+      currentUser: null,
+      input: '',
+      dropdownValue: 'all',
+      ratingValue: 'any'
     }
   }
   
   componentDidMount = () => {
     fetchMovies()
     .then(allMovies => this.setState({ movies: allMovies.movies}))
+    .then(() => this.findMovieGenres())
     .catch(error => this.setState({ error: error.message }))
   }
 
+  findMovieGenres = () => {
+    this.state.movies.map(movie => {
+      fetchSingleMovie(movie.id)
+      .then(singleMovie => movie.genres = singleMovie.movie.genres)
+    })
+  }
```

Comment:
I know fetching all the single movies upon page load to get their genres this isn't the ideal way of handling this but this was a good work around for now. This could be a potential area for refactor in the future.

Reply:
This is a great point. I'd love to explore if we could place this logic in a componentDidUpdate.